### ISSUE TITLE
Cleaner implementation of get_raw_adc_data in roach2_interface

### ISSUE THREAD

**Issue by wcpettus** [project8/dragonfly#136]
In the `get_raw_adc_data` (also `get_T_packets` and `get_F_packets`) method of the Roach2Interface, it dumps the data into a json file.  If an operator is creating multiple such dumps, there is no mechanism to create a unique directory for the data.

```python
        filename = '{}/mean_of_{}_raw_adc_spectra.json'.format(path, str(16*N))
        with open(filename, 'w') as outfile:
            json.dump(p.tolist(), outfile)
```

Should we just make mdreceiver write the data?

**Comment by cclaessens** [comment]
The filepath and name are not preset anymore. If no filename is given the data is returned instead of written. This is still not ideal because I think that when called from claude by the operator, trying to write the data on Zeppelin results in permission denied. Need to verify this.

**Comment by wcpettus** [comment]
fixed in v1.12.2
